### ISSUE TITLE
Heroic opens wine explorer window after clicking play with game porting toolkit

### ISSUE THREAD

**Issue by blackxfiied** [Heroic-Games-Launcher/HeroicGamesLauncher#2858]
### Describe the bug

I know the problem, it's becuase of the --wrapper argument. instead of referencing the "gameportingtoolkit" script in /usr/local/bin, heroic references gameportingtoolkit's wine64 (wine 7.7)

here's what heroic tries to execute according to logs (game launched is pikuniku.)
LD_PRELOAD= /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch 41869934302e4b8cafac2d3c0e7c293d --language en --wrapper "/usr/local/Cellar/game-porting-toolkit/1.0.2/bin/wine64 /Users/blackxfiied/my-game-prefix" --no-wine

this wont work becuase its trying to reference a folder to wine, which will just open the folder in file explorer

what does work is
LD_PRELOAD= /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary launch 48619f5656da446abccb727b78cda644 --language en --wrapper "/usr/local/bin/gameportingtoolkit /Users/blackxfiied/my-game-prefix" --no-wine

becuase the game poprting toolkit script executes it instead of feeding it directly to wine

### Add logs

```text
(10:25:56) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary --version
(10:25:56) INFO:    [Legendary]:        Legendary location: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary
(10:25:56) INFO:    [Gog]:              GOGDL location: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl
(10:25:56) INFO:    [Connection]:       Connectivity: check-online
(10:25:56) INFO:    [Connection]:       Pinging external endpoints
(10:25:56) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2557.0"
  }
}
(10:25:56) WARNING: [Backend]:          Protocol already registered.
(10:25:57) INFO:    [Backend]:          Loading Screen Ready
(10:25:57) INFO:    [Frontend]:         Refreshing undefined Library
(10:25:57) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(10:25:57) INFO:    [Backend]:          Frontend Ready
(10:25:57) INFO:    [Backend]:          Checking for new Heroic Updates
(10:25:57) INFO:    [Backend]:          Checking for current version changelog
(10:25:57) INFO:    [Connection]:       Connectivity: online
(10:25:57) ERROR:   [Gog]:              Unable to syncQueued playtime, userData not present
(10:25:57) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(10:25:59) INFO:    [GlobalConfig]:     Searching for Gaming Porting Toolkit Wine
(10:25:59) INFO:    [GlobalConfig]:     Found Gaming Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.2/bin
(10:25:59) INFO:    [Winetricks]:       Downloading Winetricks
(10:26:01) WARNING: [Backend]:          Could not determine System Info
(10:26:02) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/blackxfiied/Library/Application Support/heroic/gog_store/auth.json" --version
(10:26:02) INFO:    [Legendary]:        Getting more details with 'legendary info'
(10:26:02) DEBUG:   [Backend]:          {"free":122620157952,"diskSize":469665579008,"message":"114.2 GiB / 437.41 GiB","validPath":true}
(10:26:02) INFO:    [GlobalConfig]:     Searching for Gaming Porting Toolkit Wine
(10:26:02) INFO:    [GlobalConfig]:     Found Gaming Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.2/bin
(10:26:02) INFO:    [Backend]:          Starting the Download Queue
(10:26:02) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary info Sugar --platform Windows --json
(10:26:08) INFO:    [GlobalConfig]:     Searching for Gaming Porting Toolkit Wine
(10:26:08) INFO:    [GlobalConfig]:     Found Gaming Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.2/bin
(10:26:08) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary --version
(10:26:08) INFO:    [Gog]:              Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/gogdl --auth-config-path "/Users/blackxfiied/Library/Application Support/heroic/gog_store/auth.json" --version
(10:26:09) INFO:    [GlobalConfig]:     Searching for Gaming Porting Toolkit Wine
(10:26:09) INFO:    [GlobalConfig]:     Found Gaming Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.2/bin
(10:26:14) INFO:    [Legendary]:        Getting more details with 'legendary info'
(10:26:14) DEBUG:   [Backend]:          {"free":122539143168,"diskSize":469665579008,"message":"114.12 GiB / 437.41 GiB","validPath":true}
(10:26:14) INFO:    [GlobalConfig]:     Searching for Gaming Porting Toolkit Wine
(10:26:14) INFO:    [GlobalConfig]:     Found Gaming Porting Toolkit Wine at /usr/local/Cellar/game-porting-toolkit/1.0.2/bin
(10:26:15) INFO:    [Legendary]:        Running command: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary info Sugar --platform Windows --json
(10:26:26) DEBUG:   [Backend]:          {"free":122561220608,"diskSize":469665579008,"message":"114.14 GiB / 437.41 GiB","validPath":true}
(10:26:50) INFO:    [Backend]:          Writing config for Sugar
(10:26:50) DEBUG:   [Legendary]:        Using cached install info
(10:26:50) INFO:    [DownloadManager]:  Rocket League®  was added to the download queue.
(10:26:50) INFO:    [Backend]:          Preventing machine to sleep
(10:26:51) DEBUG:   [Legendary]:        Using cached install info
(10:26:51) INFO:    [Legendary]:        Installing Sugar: /Applications/Heroic.app/Contents/Resources/app.asar.unpacked/build/bin/darwin/legendary install Sugar --platform Windows --base-path /Users/blackxfiied/Games/Heroic --skip-dlcs --skip-sdl -y
(10:26:51) DEBUG:   [Legendary]:        Logging to file "/Users/blackxfiied/Library/Application Support/heroic/GamesConfig/Sugar.log"
(10:27:00) INFO:    [Legendary]:        Progress for Rocket League®: 0%/0.00MB/00:00:00 Down: 0MB/s / Disk: 0MB/s

bruh im downloading rocket league but its required to add logs
```

### Steps to reproduce

Press play on any game while 'GPTK Wine (DX11/DX12 Only) - wine7.7' is selected as your Crossover/Wine version

### Expected behavior

The game launches.

### Heroic Version

Latest Stable

### System Information

macOS 14.0 beta 3
Apple MacBook Pro 14", 2021
Apple M1 Pro APU
Heroic 2.8.0 (that's the newest release, not 2.9.0?)

**Comment by flavioislima** [comment]
First, this feature is not out yet.
Second, you did not add the logs from the game launching so it's hard to know what the command is doing.
Third, Heroic won't use the script, it will find the wine64 binary and pass the wine to legendary without the wrapper.

So wait for heroic 2.9 is out before opening an issue related with that. 👍🏽

**Comment by blackxfiied** [comment]
> Second, you did not add the logs from the game launching so it's hard to know what the command is doing.
all the command is doing that's important is what I attached

> Third, Heroic won't use the script, it will find the wine64 binary and pass the wine to legendary without the wrapper.
then why doesn't heroic do it properly?

i was told 2.9 was coming out today or tomorrow
just letting you know in advance becauase i don't want a situation where this becomes a problem that us as the consumers have to deal with, why not just get it over and done with to begin with?

So thanks, i hope 2.9 fixes the issue related with that. 👍🏽

**Comment by Git-North** [comment]
A PR would be great if you have a working code

**Comment by arielj** [comment]
Can anyone in this thread confirm if this got fixed in Heroic 2.9.1?

**Comment by arielj** [comment]
Closing this since there's no answer since my question in July
